<launch>
    <rosparam param="use_sim_time">false</rosparam>

    <arg name="radius_comm" value="5.0" />
    <arg name="ball_radius_multiplier" value="1.5" />

    <include file="$(find simulator)/launch/entity_controller.launch">
        <!--Determine generation rule of UAVs and targets-->
        <arg name="radius_comm" value="$(arg radius_comm)"/>
    </include>

    <include file="$(find optimizer)/launch/star_convex.launch">
        <!--Determine parameters of star convex calculation and visualization-->
        <arg name="ball_radius_multiplier" value="$(arg ball_radius_multiplier)"/>
    </include>

    <include file="$(find simulator)/launch/episode_controller.launch">
        
    </include>

    <arg name="map_size_x" value="20.0"/>
    <arg name="map_size_y" value="20.0"/>
    <arg name="map_size_z" value="10.0"/>
    <include file="$(find map_generator)/launch/map_generator.launch">
        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <!-- Number of circle and pillar obstacles -->
        <arg name="c_num" value="0"/>
        <arg name="p_num" value="10"/>

        <!-- <arg name="odometry_topic" value="$(arg odom_topic)" /> -->
    </include>

    <node name="steiner_tree" pkg="algorithm" type="steiner_tree" output="screen">
        <param name="graph_constructor/x_size" value="$(arg map_size_x)" type="double"/>
        <param name="graph_constructor/y_size" value="$(arg map_size_y)" type="double"/>
        <param name="graph_constructor/z_size" value="$(arg map_size_z)" type="double"/>
        <param name="grid_map/resolution" value="2.0" type="double"/>

        <remap from="/global_map" to="/map_generator/inflated_global_cloud"/>
        <remap from="~uav_num" to="uav_num"/>
    </node>
    
    <node pkg="optimizer" name="optimizer" type="optimizer" output="screen">
        <param name="optimizer/approximation_quality" value="50.0" type="double"/>
        <param name="optimizer/safe_margin" value="2.0" type="double"/>
        <param name="optimizer/penalty_weight" value="80000.0" type="double"/>
        <param name="optimizer/radius_comm" value="$(arg radius_comm)" type="double"/>
        <param name="optimizer/ball_radius_multiplier" value="$(arg ball_radius_multiplier)" type="double"/>
        <!-- Should read from appropriate locations in the future -->
        <param name="optimizer/init_x" value="0.0" type="double"/>
        <param name="optimizer/init_y" value="0.0" type="double"/>
        <param name="optimizer/init_z" value="5.0" type="double"/>
        <param name="optimizer/tar_num" value="3" type="int"/>
    </node>

</launch>